### PR TITLE
SB-21591 Update the current Self declaration form to the new Consent form

### PR DESCRIPTION
**Pull request by reshmi-nair**
…form

### CODE REVIEW

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserSelfDeclarationManagementActor.java [hunk 1]
```diff
@@ -1,299 +1,140 @@
...
-    }
-    if (UserActorOperations.UPDATE_USER_SELF_DECLARATIONS_ERROR_TYPE
-        .getValue()
-        .equalsIgnoreCase(request.getOperation())) {
-      updateUserSelfDeclaredErrorStatus(request);
-    } else {
-      onReceiveUnsupportedOperation("upsertUserSelfDeclarations");
+    String operation = request.getOperation();
+    switch (operation) {
+      case "updateUserDeclarations": // update self declare
```
Comment (Hari-stackroute) [inline actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserSelfDeclarationManagementActor.java:34]: Retain with the constants in the switch cases

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/dao/impl/UserSelfDeclarationDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  public void insertSelfDeclaredFields(Map<String, Object> extIdMap, RequestContext context) {
+
+    cassandraOperation.insertRecord(
+        JsonKey.SUNBIRD, JsonKey.USER_DECLARATION_DB, extIdMap, context);
+  }
+
+  public List<Map<String, Object>> getUserSelfDeclaredFields(
+      UserDeclareEntity userDeclareEntity, RequestContext context) {
+    List<Map<String, Object>> dbResExternalIds = new ArrayList<>();
```
Comment (Hari-stackroute) [inline actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/dao/impl/UserSelfDeclarationDaoImpl.java:40]: Please add loggers for the entry and exit.

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/impl/UserSelfDeclarationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  public static UserSelfDeclarationService getInstance() {
+    if (selfDeclarationService == null) {
+      selfDeclarationService = new UserSelfDeclarationServiceImpl();
+    }
+    return selfDeclarationService;
+  }
+
+  public Response saveUserSelfDeclareAttributes(
+      Map<String, Object> requestMap, RequestContext context) {
+    List<UserDeclareEntity> selfDeclaredFields =
```
Comment (Hari-stackroute) [inline actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/impl/UserSelfDeclarationServiceImpl.java:42]: Please consider about legs.

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserSelfDeclarationManagementActor.java (hunk 1 above)
Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserSelfDeclarationManagementActor.java:34]: Cannot retain constants as ActorOperations is an enum class. All the actors follow same logic as of now.

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/dao/impl/UserSelfDeclarationDaoImpl.java (hunk 2 above)
Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/dao/impl/UserSelfDeclarationDaoImpl.java:40]: Entry and exit logs has to be in API level,  so another ticket will take care of this for all APIs

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/impl/UserSelfDeclarationServiceImpl.java (hunk 3 above)
Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/impl/UserSelfDeclarationServiceImpl.java:42]: Entry and exit logs has to be in API level, so another ticket will take care of this for all APIs

**Review by Hari-stackroute — APPROVED** [review]
This pr looks fine for Default persona, but we need to handle for old personas to remove from system.

**Pull request merged**